### DISCUSSION TITLE
Koishi 4.13.1

### DISCUSSION THREAD

**Discussion by shigma** [koishijs/koishi#1099] [General]
## Notable Changes

- 新增了主题系统，允许插件定义新的主题 (koishijs/webui@856f49cc7ecd94f8d8fe814137a1a07bb63efc16, koishijs/webui@86eb4b29454a11f0da45df12679c68a2b6d4f269, koishijs/webui@add178b4cfff26597fd2fab2f0ffc322a7ef7827, koishijs/webui@47363ebf2f29eb236663333319c5289ba0f503a5, koishijs/webui@677d71d03866a2af4203df836d64715a4abf21dd, koishijs/webui@d9565e474a350b8d13ad3de0dfec3d8df0688601)
- 将用户设置拆分为独立的子路由，并允许插件扩展配置项 (koishijs/webui@5614aaa7c82c3a5ff80e202c1423099fa07f7fcd, koishijs/webui@5dc95d76124344f8c1cc815d10e283eb2507b37a, koishijs/webui@0a3039346cf6946b91f591fa79423526d2f230b4)

## Features

- **explorer:** 文件选择对话框的左上角增加了用于回退到上一级目录的按钮 (koishijs/webui@4514ebf8868fffb446020772adf7e6ffd8324684)
- **explorer:** 为文件选择对话框中的文件、目录增加了图标提示 (koishijs/webui@6897219acc80812309f580fbe5aab0f70b4484bb)
- **explorer:** `Schema.path()` 支持了 `allowCreate` 选项，当开启时将允许在对话框内创建目录、上传文件 (koishijs/webui@e2f9a6e315a593085b16f911a416e362ecf4a643)
- **explorer:** `Schema.path()` 支持了 `filters` 选项，可以指定选择的对象是文件、目录、还是具体的文件类型 (通过后缀名区分)；指定类型后会对可选的文件、目录进行过滤 (koishijs/webui@4071998c0f345a698c9adcc04842fadea0fe178d)
- **insight:** 优化了交互时的高亮效果 (koishijs/webui@fbbcd47b41331f33b809ee99fdcea35abcb813b2)
- **insight:** 为插件组和服务类插件增加了特殊表示 (koishijs/webui@ee73c78bc839c48ccf893b4eb7cf281b8b0345d0)
- **schema:** 实验性地支持了 options.autofix (shigma/schemastery@323980ff1c35966d5e5101fdb08f21587e5d659c)

## Bug Fixes

- **client:** 修复了部分页面底部存在遮挡的问题 (koishijs/koishi#1097) (koishijs/webui@1e38cb8afb8b8f7c11904315d94c7332c1d8c6bf)
- **core:** 修复了在指令回调中返回 `h.i18n()` 无法使用路径前缀的问题 (koishijs/koishi#1095) (koishijs/koishi@3e871850912dc19f3ce8eec689846fdf6d489be3)
- **explorer:** 修复了编辑器主题不随控制台主题切换的问题 (koishijs/webui@214f21e6eb52c54706832c9da8b195a08bf1159f)

## Other Changes

- chore: bump versions (koishijs/koishi@e92e48011c09333c584069199b07382f400dcfb4)
- refa: move theme default to console/app (koishijs/webui@61c3e006287007f820a7c1e14ed04f26c804d2a0)
- chore: bump versions (koishijs/webui@d807a222569268773dd4594ce16374950cea9912)
- chore: enhance class names (shigma/schemastery@93dbaa0fd9be933cff626913d0a0c48e9ab5fbab)
- refa: enhance exported apis (shigma/schemastery@ca3560c9d1dba6b47d5703e49cbf244f628951f0)
- refa: move back app to client package (koishijs/webui@438b2dd2800a513006ba67d0ad542972d3b85558)
- refa: enhance schema components (koishijs/webui@b5b95601f7d8f58aa09356fa0a72ad604a103c38)
- chore: bump versions (koishijs/webui@93f13412df710e8f49e6dce4378a73a57d88c0e1)
- chore: bump versions (satorijs/satori@9530520c226d1f01188e420ad20fcf577b936d53)
- chore: bump versions (koishijs/koishi@ad71c0d0cd533ca63a83eba54132c187cbc4bd9b)

<hr /><em>This discussion was created from the release <a href='https://github.com/koishijs/koishi/releases/tag/4.13.1'>Koishi 4.13.1</a>.</em>

**Comment by maxoyed** [comment]
想使用自己通过 Verdaccio 搭建的registry，配置好之后，插件市场提示“您所用的 registry 不支持搜索功能 (如 npmmirror)，请考虑进行更换”。请问这块对 registry 有什么具体的要求么，比如在 `search.endpoint` 中配置的 URL 应该支持什么格式的输入输出？

**Reply by ilharp** [reply]
与 npm Registry 的搜索 API 要求相同。
